calcom/cal.com
# feat: [CAL-1816] Organization subdomain support

## What does this PR do?

Implementing the changes in middleware to support subdomains.

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

```
MRW = Middleware ReWrite			null = non-existing org slug
CRW = Config ReWrite				acme = existing org slug
PSS = Page getServerSideProps			alex = existing user slug in acme
PSSR = PSS Redirect				peter = non-existing user in acme
						engineering = existing team in acme
						foobar = non-existing team in acme

--- org.cal.local:3000

acme.cal.local:3000 
	-(MRW)-> acme.cal.local:3000/org/acme 
	-(CRW)-> acme.cal.local:3000/team/acme
	-(PSS: check acme is an org)-> Shows Acme profile

null.cal.local:3000 
	-(MRW)-> acme.cal.local:3000/org/null 
	-(CRW)-> acme.cal.local:3000/team/null 
	-(PSS: check null is an org)-> Not an org 
	-(PSSR)-> 404

acme.cal.local:3000/alex 
	-(MRW: check alex is team in acme)-> Non-team
	-(PSS: check alex belongs to org acme)-> Belongs
	-(PSS)-> Shows alex profile

acme.cal.local:3000/peter 
	-(PSS: check peter is team in acme)-> Non-team
	-(PSS: check peter belongs to org acme)-> Doesn't belong
	-(PSSR)-> 404

acme.cal.local:3000/engineering 
	-(MRW: check engineering is team in acme)-> Is team
	-(MRW)-> acme.cal.local:3000/team/engineering 
	-(PSS)-> Shows engineering team profile

acme.cal.local:3000/foobar
	-(MRW: check foobar is team in acme)-> Non-team
	-(PSS: check foobar belongs to org acme)-> Doesn't belong
	-(PSSR)-> 404

--- cal.local:3000
[TODO]

```
